### ISSUE TITLE
replace missing value by average of the value (over all molecules for which it is not missing)

### ISSUE THREAD

**Issue by UnixJunkie** [mordred-descriptor/mordred#53]
Hello,

This is a feature request.

I compute all descriptors for a dataset.

Sometimes, a mol. desc. value is missing.

It manifests as a missing column value in the output file (i.e. two consecutive comas).
It would be nice (if all molecules can be loaded into memory) to replace
such values by the average value for that column (N/A excluded).
It the value is missing for the whole column, then put a 0 for the whole column.

Also, there are some True or False string values in the output csv file.
I guess they should be replaced by 1 and 0, respectively.

Thanks a lot for mordred,
Francois.

**Comment by philopon** [comment]
Thank you for your feedback.

> It manifests as a missing column value in the output file (i.e. two consecutive comas).
> It would be nice (if all molecules can be loaded into memory) to replace
> such values by the average value for that column (N/A excluded).
> It the value is missing for the whole column, then put a 0 for the whole column.

In machine learning, the average value to interpolate is calculated by only training dataset to avoid leakage. To solve this, save mean values or provide training set in dataset. Both approaches are complex.

I think this is the problem that should solve by machine learning library (e.g. Imputer of scikit-learn).

> Also, there are some True or False string values in the output csv file.
> I guess they should be replaced by 1 and 0, respectively.

It is reasonable. I opened new issue (#54).

Best regards,

**Comment by UnixJunkie** [comment]
Now, I do realize the fix is not so easy, unless if we consider all molecules are the training or the test set.
In that case, we could output the mean values in some file that the user would reuse in combination with the test set.
For example, when encoding the training set
```
--train-means-out means.csv
```
would be written out.
When encoding the test set
```
 --train-means-in means.csv
```
would be read in and the values used to fill in missing values.

**Comment by philopon** [comment]
Well... Of course, we can implement this feature.
However, I think that the benefit of the feature is less than its complexity.

Why don't you use machine learning libraries instead of the feature?

Best regards,

**Comment by UnixJunkie** [comment]
Not all ML libraries do this kind of missing values management.
But ok, don't bother, I will handle it on my side.

**Comment by philopon** [comment]
I'm sorry I couldn't be of any help.
